astroquery/noao/core.py

```diff
@@ -0,0 +1,70 @@
+"""
+Provide astroquery API access to OIR Lab Astro Data Archive (natica).
+
+This does DB access through web-services.
+"""
+
+import astropy.table
+from ..query import BaseQuery
+from ..utils import async_to_sync
+from ..utils.class_or_instance import class_or_instance
+from . import conf
+
+
+__all__ = ['Noao', 'NoaoClass']  # specifies what to import
+
+
+@async_to_sync
+class NoaoClass(BaseQuery):
+
+    NAT_URL = conf.server
+    ADS_URL = f'{NAT_URL}/api/adv_search/fasearch'
+    SIA_URL = f'{NAT_URL}/api/sia/voimg'
+
+    def __init__(self, which='voimg'):
+        """ set some parameters """
+        self._api_version = None
+
+        if which == 'vohdu':
+            self.url = f'{self.NAT_URL}/api/sia/vohdu'
+        if which == 'voimg':
+            self.url = f'{self.NAT_URL}/api/sia/voimg'
+        else:
+            self.url = f'{self.NAT_URL}/api/sia/voimg'
+
+        super().__init__()
+
+    @property
+    def api_version(self):
+        if self._api_version is None:
+            response = self._request('GET',
+                                     f'{self.NAT_URL}/api/version',
+                                     cache=True)
```

Comment:
this is a case where I suspect you do _not_ want to cache, since you want to check the remote API version each time you query.  I'm not sure, though.

Reply:
Nope.  As long as someone doesn't create an instance of NOAO class and keep it up for days, it will be fine.  Even then, unlikely to get tripped up.